Physical_Engine/Query/GeometricalCentreline.cs

```diff
@@ -0,0 +1,64 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Geometry;
+using BH.Engine.Spatial;
+using BH.oM.Geometry;
+using BH.oM.Physical.Elements;
+using BH.oM.Physical.FramingProperties;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Spatial.ShapeProfiles;
+using System.ComponentModel;
+
+namespace BH.Engine.Physical
+{
+    public static partial class Query
+    {
+        [Description("Returns the geometrical centreline of an IFramingElement.")]
+        [Input("element", "The IFramingElement to query the geometrical centreline of.")]
+        [Output("curve", "The geometrical centreline of the IFramingElement.")]
+        public static ICurve GeometricalCentreline(this IFramingElement element)
+        {   
+            ICurve location = element.Location;
+
+            if (element.Property is ConstantFramingProperty)
+            {
+                ConstantFramingProperty constantProperty = element.Property as ConstantFramingProperty;
+
+                IProfile profile = constantProperty.Profile; 
+
+                Point geometricalCentre = profile.Edges.Bounds().Centre();
```

Comment:
You could do it in one line, then it would be good to add a null check:
```suggestion
                Point geometricalCentre = (element.Property as ConstantFramingProperty)?.Profile?.Edges?.Bounds()?.Centre();
                if (geometricalCentre == null)
                    return null;
```

Alternatively, you can keep the code structure you have but you would need to add a null check after every line.